Спринт 23

il76 opened the pull request.

Commit:
Версии компонентов.
Иконки.
Раскомментировал текствотчер для второго поля на добавлении плейлиста.

Commit:
Экран для плейлистов и загрузка данных плейлиста на нём

Commit:
На странице плейлиста вывод названия и описания из БД

Commit:
Выводим длительность плейлиста и количество треков в плейлисте.
В таблицу трекоа добавлена колонка - флаг, является ли трек избранным

Commit:
Второй Bottom Sheet с информацией о плейлисте и пока неактивными кнопками

Commit:
Шаринг

Commit:
Починил шаринг текста

Commit:
Шаринг при пустом списке.
Прячем BottomSheet с треками, если их нет

Commit:
Удаление плейлиста и выход на предыдущий экран

Commit:
Могли бы и в дизайне нарисовать состояние, когда нет треков, а не в самом конце задания текстом написать.

Commit:
Удаление трека из плейлиста и, при необходимости, из списка треков.

Commit:
Методы обновления плейлиста

Commit:
Эксперименты с формой редактирования плейлиста

Commit:
Потерял параметр во фрагменте

Commit:
Выводим треки на странице плейлиста в порядке добавления DESC

Commit:
Решение проблем с вёрсткой при возврате из редактирования в плейлист

Commit:
Редактирование плейлиста

Commit:
Редактирование плейлиста

il76 commented:
В коде есть определённое количество копипасты, но отрефакторить я уже не успевал.

anastasianadelshtehel commented on app/src/main/java/com/il76/playlistmaker/playlist/ui/PlaylistViewModel.kt:
```diff
@@ -0,0 +1,104 @@
+
+    fun loadTracks() {
+        viewModelScope.launch {
+            tracksInteractor.getPlaylistTracks(playlist!!.id).collect { tracksData ->
```
Лучше использовать безопасные вызовы (?.) или проверки на null

anastasianadelshtehel commented on app/src/main/java/com/il76/playlistmaker/playlist/ui/PlaylistViewModel.kt:
```diff
@@ -0,0 +1,104 @@
+    }
+
+    fun sharePlaylist() {
+        var share_text = "В этом плейлисте нет списка треков, которым можно поделиться"
+        if (!tracksList.isNullOrEmpty() && playlist != null) {
+            share_text = playlist?.name + "\n" + playlist?.description + "\n" + tracksList?.size + " треков" + "\n\n"
```
Лучше вынести строки в resources/strings.xml

anastasianadelshtehel commented on app/src/main/java/com/il76/playlistmaker/playlist/ui/PlaylistFragment.kt:
```diff
@@ -0,0 +1,256 @@
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = FragmentPlaylistBinding.inflate(inflater, container, false)
```
Можно использовать более краткую форму: `FragmentPlaylistBinding.inflate(inflater).root`

anastasianadelshtehel commented on app/src/main/java/com/il76/playlistmaker/playlist/ui/PlaylistViewModel.kt:
```diff
@@ -0,0 +1,104 @@
+    }
+
+    fun loadPlaylist() {
+        viewModelScope.launch {
```
loadPlaylist() дублирует код из init

Commit:
Починил удаление треков, сломавшееся после внедрения идентификатора в таблицу связей.
Переименовал кнопки в диалоге, хотя в задании были написаны именно такие формулировки.

Commit:
Мелкие правки

Commit:
Починил добавление уже существующего трека в избранное.

Commit:
Отступы.
Конфиг
